pyoxidizer/src/py_packaging/distribution.rs

```diff
@@ -360,24 +475,170 @@ impl ParsedPythonDistribution {
     }
 
     /// Ensure pip is available to run in the distribution.
-    pub fn ensure_pip(&self) -> PathBuf {
-        let pip_path = self
-            .python_exe
-            .parent()
-            .expect("could not derive parent")
-            .to_path_buf()
-            .join(PIP_EXE_BASENAME);
+    pub fn ensure_pip(&self, logger: &slog::Logger) -> PathBuf {
+        let dist_prefix = self.base_dir.join("python").join("install");
+        let python_paths = resolve_python_paths(&dist_prefix, &self.version);
+
+        let pip_path = python_paths.bin_dir.join(PIP_EXE_BASENAME);
 
         if !pip_path.exists() {
-            std::process::Command::new(&self.python_exe)
-                .args(&["-m", "ensurepip"])
-                .status()
-                .expect("failed to run ensurepip");
+            warn!(logger, "{} doesnt exist", pip_path.display().to_string());
+            invoke_python(&python_paths, &logger, &["-m", "ensurepip"]);
         }
 
         pip_path
     }
 
+    /// Duplicate the python distribution, with distutils hacked
+    pub fn create_hacked_base(&self, logger: &slog::Logger) -> PythonPaths {
+        let venv_base = self.venv_base.clone();
+
+        let venv_dir_s = self.venv_base.display().to_string();
+
+        if !venv_base.exists() {
+            if self.os == "windows" {
+                let external_python = which("python").unwrap().clone();
+                warn!(logger, "python={}", external_python.display().to_string());
+                // TODO: check python version
+
+                let external_dist_prefix = external_python.parent().unwrap();
+
+                copy_dir(&external_dist_prefix, &venv_base).unwrap();
+
+                let external_dist_prefix_s = external_dist_prefix.display().to_string();
+                warn!(
+                    logger,
+                    "copied {} to create hacked base {}", external_dist_prefix_s, venv_dir_s
+                );
+
+                let dist_prefix = self.base_dir.join("python").join("install");
+
+                let static_base = self.base_dir.parent().unwrap().join("static_base");
```

Comment:
this static_base isnt used.  I was using it to run `setup.py build_ext` twice, once for the pip build environment and the second for the static libs, but so far it seems that compiling the object files twice is sufficient.